web/src/engine/websites/ModeScanlator.ts

```diff
@@ -1,22 +1,76 @@
 import { Tags } from '../Tags';
 import icon from './ModeScanlator.webp';
-import { DecoratableMangaScraper } from '../providers/MangaPlugin';
-import * as MangaStream from './decorators/WordPressMangaStream';
+import { type Chapter, DecoratableMangaScraper, Page } from '../providers/MangaPlugin';
 import * as Common from './decorators/Common';
+import { Fetch, FetchRequest, FetchWindowScript } from '../FetchProvider';
+import * as JSZip from 'jszip';
 
-@MangaStream.MangaCSS(/^https?:\/\/modescanlator\.com/)
-@Common.MangasSinglePageCSS('/projetos/', 'div.page div.bs div.bsx > a', Common.AnchorInfoExtractor(true))
-@MangaStream.ChaptersSinglePageCSS()
-@MangaStream.PagesSinglePageCSS()
+const pagescript = `
+    new Promise((resolve, reject) => {
+        try {
+            resolve(urls);
+        } catch (error) {
+            const pages = [...document.querySelectorAll('div#imageContainer img')].map(image=> new URL(image.src, window.location.origin).href);
+            resolve(pages);
+        }
+    });
+`;
+
+function ChapterExtractor(anchor: HTMLAnchorElement) {
+    const id = anchor.pathname;
+    const title = anchor.querySelector('div.info__capitulo__obras span.numero__capitulo').textContent.trim();
+    return { id, title };
+}
+
+@Common.MangaCSS(/^https?:\/\/modescanlator\.com\/[^/]+\/$/, 'h1.desc__titulo__comic')
+@Common.MangasSinglePageCSS('/todas-as-obras/', 'div.comics__all__box a.titulo__comic__allcomics')
+@Common.ChaptersSinglePageCSS('ul.capitulos__lista a.link__capitulos', ChapterExtractor)
 @Common.ImageAjax()
 
 export default class extends DecoratableMangaScraper {
 
     public constructor() {
         super('modescanlator', `Mode Scanlator`, 'https://modescanlator.com', Tags.Language.Portuguese, Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Source.Scanlator);
     }
-
     public override get Icon() {
         return icon;
     }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
```

Comment:
The method `FetchPages` is supposed to be lightweight on resources.
If there are many images, this method will run for a long time (progress bar not moving) and may use a lot of memory.

I suggest this method to return each page with the link to the zip archive and the filename which shall be extracted.
Duplicated zip archive downloads are probably handled by the browser cache.

The `FetchImage(page: Page)` method needs to be implemented accordingly to download he image from the zip archive.

Reply:
I knew you would have said that. My goal was to avoid redownloading the zips for each Page. Since i have the zip buffers in memory how  do you suggest that FetchImage can properly use them ? I feel like page parameters is a bad idea but idk.

Reply:
Just re-download for now. Keeping HaruNeko's intended concepts is more important right now than optimizing bandwidth usage.

It could be possible that the archive hosted by CloudFlare supports ranged requests, so in the future it might be possible to only extract the bytes from the zip archive needed for a certain image (after reading the zip index).

Reply:
Turns out zips are not redownloaded 👍 
![image](https://github.com/manga-download/haruneko/assets/10671518/b8b46afb-82ed-4364-8103-56982560dee4)